impeller/renderer/backend/gles/blit_command_gles.cc

```diff
@@ -117,6 +119,168 @@ bool BlitCopyTextureToTextureCommandGLES::Encode(
   return true;
 };
 
+namespace {
+struct TexImage2DData {
+  GLint internal_format = 0;
+  GLenum external_format = GL_NONE;
+  GLenum type = GL_NONE;
+  BufferView buffer_view;
+
+  explicit TexImage2DData(PixelFormat pixel_format) {
+    switch (pixel_format) {
+      case PixelFormat::kA8UNormInt:
+        internal_format = GL_ALPHA;
+        external_format = GL_ALPHA;
+        type = GL_UNSIGNED_BYTE;
+        break;
+      case PixelFormat::kR8UNormInt:
+        internal_format = GL_RED;
+        external_format = GL_RED;
+        type = GL_UNSIGNED_BYTE;
+        break;
+      case PixelFormat::kR8G8B8A8UNormInt:
+      case PixelFormat::kB8G8R8A8UNormInt:
+      case PixelFormat::kR8G8B8A8UNormIntSRGB:
+      case PixelFormat::kB8G8R8A8UNormIntSRGB:
+        internal_format = GL_RGBA;
+        external_format = GL_RGBA;
+        type = GL_UNSIGNED_BYTE;
+        break;
+      case PixelFormat::kR32G32B32A32Float:
+        internal_format = GL_RGBA;
+        external_format = GL_RGBA;
+        type = GL_FLOAT;
+        break;
+      case PixelFormat::kR16G16B16A16Float:
+        internal_format = GL_RGBA;
+        external_format = GL_RGBA;
+        type = GL_HALF_FLOAT;
```

Comment:
I don't think these are supported by GL ES 2.0 but we will probably put these behind a capability check anyway I suppose.